ir/src/boundary_stmts.rs

```diff
@@ -0,0 +1,175 @@
+use crate::TraceSegment;
+
+use super::{BTreeMap, BoundaryExpr, IdentifierType, SemanticError, SymbolTable};
+use parser::ast::{self, BoundaryStmt};
+
+// BOUNDARY CONSTRAINTS
+// ================================================================================================
+
+/// A struct containing all of the boundary constraints to be applied at each of the 2 allowed
+/// boundaries (first row and last row). The constraints are stored in a vector of tuples where
+/// each index of the vector corresponds to the trace segment of the column the boundary constraint
+/// applies to. For ease of code generation and evaluation, constraints are sorted into maps by the
+/// boundary. This also simplifies ensuring that there are no conflicting constraints sharing a
+/// boundary and column index.
+#[derive(Default, Debug)]
+pub(crate) struct BoundaryStmts {
+    boundary_constraints: Vec<(BTreeMap<usize, BoundaryExpr>, BTreeMap<usize, BoundaryExpr>)>,
```

Comment:
I think this approach makes sense for the trace segment handling. Wdyt about defining a type alias for `BTreeMap<usize, BoundaryExpr>`?

Reply:
I've defined an alias with the name `BoundaryConstraintsMap`. Please let me know if you have a better name suggestion.